Sources/XcodeProjToGraph/ProcessRunner/Lipo/LipoArchsResult.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+@preconcurrency import XcodeGraph
+
+/// Represents the architectures extracted from running `lipo -archs`.
+///
+/// `LipoArchsResult` provides a structured result indicating which architectures
+/// are contained in a given binary. It uses `XcodeGraph.BinaryArchitecture` to
+/// represent each architecture (e.g., arm64, x86_64).
+public struct LipoArchsResult: Sendable {
+    /// The list of architectures reported by `lipo -archs`.
+    public let architectures: [XcodeGraph.BinaryArchitecture]
+}
+
+/// Parses the output of a `ProcessResult` from a `lipo -archs` invocation into a `LipoArchsResult`.
+///
+/// `lipo -archs` typically returns a single line of whitespace-separated architectures,
+/// for example: `"arm64"` or `"arm64 x86_64"`.
+///
+/// - Parameter result: The `ProcessResult` obtained from running the `lipo -archs` command.
+/// - Returns: A `LipoArchsResult` containing all detected architectures.
+/// - Throws:
+///   - `ProcessRunnerError.failedToRunProcess` if `lipo` exits with a non-zero code.
+///   - Any errors related to decoding or interpreting the output as architectures.
+public func parseLipoArchsResult(_ result: ProcessResult) throws -> LipoArchsResult {
```

Comment:
Have you considered using the implementation that we have in tuist: https://github.com/tuist/tuist/blob/main/Sources/TuistCore/MetadataProviders/FrameworkMetadataProvider.swift ?

I believe we can use it instead. If so, then I'd consider creating a new module called something like `ProductMetadataProvider` or `BinaryMetadataProvider` that we can then share back with `tuist` instead of having the implementation duplicated. If we end up doing this, I'd consider moving the whole `MetadataProviders` directory into `XcodeGraph`: https://github.com/tuist/tuist/tree/main/Sources/TuistCore/MetadataProviders

Reply:
That’s a great callout. Thanks for pointing this out! I’ve added TODO comments to explore how best to align with this or where it should live. It might not be as straightforward as dropping it in, since the `FileHandler` here already uses `TuistSupport` (though only for the Info.plist). That said, I think this could be a great opportunity to modularize FileHandling or maybe more as a whole, especially if we can align with the broader patterns in Tuist.

Reply:
The `FileHandler` should be removed in favor of `FileSystem`. It should be already possible. If that's the only blocking thing to move it here, then we should be good.